inline.h

```diff
@@ -3028,7 +3028,7 @@ Perl_utf8n_to_uvchr_msgs(const U8 *s,
         UV state = PL_strict_utf8_dfa_tab[256 + type];
         uv = (0xff >> type) & NATIVE_UTF8_TO_I8(*s);
 
-        while (++s < send) {
+        while (LIKELY(state != 1) && ++s < send) {
```

Comment:
[MEMO] We need to ensure that the state after the first transition isn't `1`. We expect the state to be a multiple of `19`, so referencing `PL_strict_utf8_dfa_tab` with a state of `1` is completely invalid.